### PR TITLE
Add typed property retrieval function

### PR DESCRIPTION
**Pull request by HLWeil**
Generic type annotations are not supported at runtime in js and py. But: by inlining, Fable allows the function to be transpiled against all type cases that appear throughout the transpilation. 
Inlining is only possible for functions in Fable, so instead of fixing the member, I propose this new function in the `DynObj` module.

workaround for #26

### CODE REVIEW

File: src/DynamicObj/DynamicObj.fs [hunk 1]
```diff
@@ -27,15 +27,20 @@ type DynamicObj() =
...
         | None -> None
-        | Some o -> 
+        | Some o ->      
             match o with
-            | :? 'a -> o :?> 'a |> Some
+            | :? 'a as o -> o |> Some
             | _ -> None
-
+        #else
+        failwith "Method TryGetTypedValue is not supported in Fable transpiled code. Use static method tryGetTypedValue instead."
```
Comment (kMutagene) [inline src/DynamicObj/DynamicObj.fs:42]: > Use static method tryGetTypedValue instead

the method does only exist on the `DynObj` module though, so should it not be something like 

> Method TryGetTypedValue is not supported in Fable transpiled code. Use DynObj.tryGetTypedValue instead.

?

**Comment by kMutagene** [comment]
> against all type cases that appear throughout the transpilation

So if i understand this correctly, Libraries that use this function internally will then transpile, because the types are known during transpilation, but if a consuming library wants to use this for any other types, it will still fail?

File: src/DynamicObj/DynamicObj.fs (hunk 1 above)
Comment (HLWeil) [reply src/DynamicObj/DynamicObj.fs:42]: yes!

**Comment by HLWeil** [comment]
> > against all type cases that appear throughout the transpilation
> 
> So if i understand this correctly, Libraries that use this function internally will then transpile, because the types are known during transpilation, but if a consuming library wants to use this for any other types, it will still fail?

Nope, transpiled code is not shipped. All source files, even the ones referenced from nuget, are always transpiled together.

**Comment by kMutagene** [comment]
After further discussion, we agreed on removing the `TryGetTypedValue<'T>` _in transpiled code_ completely. The reasoning here is that we have different behavior anyways (it always fails in transpiled code with the current approach).

So now, if code from this lib is to be transpiled, the inlined function `DynObj.tryGetTypedValue` must be used, otherwise transpilation will fail, meaning we move the problem from runtime to compile time, which we deemed more desirable. Otherwise, when staying in the .NET world, the instance member `TryGetTypedValue<'T>` can be used without problems.

**Review by kMutagene — APPROVED** [review]

**Pull request merged**